app/controllers/timecards_controller.rb

```diff
@@ -40,6 +40,28 @@ def update
     end
   end
 
+  def verify
+    authorize! :edit, Timecard
+    @timecard = Timecard.find(params[:timecard_id])
+    if @timecard.status == 'Verified'
+      redirect_to request.referrer || timecards_path, notice: 'Timecard already verified.'
+    else
+      @timecard.status = 'Verified'
+      if @timecard.save
+        respond_to do |format|
+          format.html {
+            redirect_to request.referrer || timecards_path
+          }
+          format.json {
+            render json: @timecard
+          }
+        end
+      else
+        render action: "edit"
```

Comment:
Where `verify`is coming in as an ajax call, is render edit the expected action ?